server/handlers.go

```diff
@@ -39,13 +39,15 @@ func (server *HTTPServer) mainEndpoint(writer http.ResponseWriter, _ *http.Reque
 
 // listOfGroups returns the list of defined groups
 func (server *HTTPServer) listOfGroups(writer http.ResponseWriter, request *http.Request) {
-	groups := make([]groups.Group, 0, len(server.Groups))
+	if server.groupsList == nil {
+		server.groupsList = make([]groups.Group, 0, len(server.Groups))
 
-	for _, group := range server.Groups {
-		groups = append(groups, group)
+		for _, group := range server.Groups {
+			server.groupsList = append(server.groupsList, group)
```

Comment:
Why do you create a copy of the array? Couldn't use just use `server.Groups`?

Reply:
I iterate over `server.Groups`, that is a `map`, to return just an array with the values (the equivalent in Python would be `server.Groups.values()`, but I don't know if there is some equivalent to it in Golang).
If I just use `server.Groups` as you suggest, the JSON response will be different as it will contain the name of each group as key, instead of a list as expected

Reply:
Ah, got it, thanks